docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1,27 +1,1957 @@
+- The call to `unlock` further checks that the key matches the locked object that was provided.
+- Instead of transferring the escrowed object to the recipient address, it is returned by the `swap` function. You can do this because you checked that the transaction sender is the recipient, and it makes this API more composable. Programmable transaction blocks (PTBs) provide the flexibility to decide whether to transfer the object as it is received or do something else with it.
+
+<details>
+<summary>
+`swap` function in `shared.move`
+</summary>
+{@inject: examples/trading/contracts/escrow/sources/shared.move#variable=EMismatchedSenderRecipient,EMismatchedExchangeObject singleSpace noComments}
+{@inject: examples/trading/contracts/escrow/sources/shared.move#fun=swap noComments noTitle}
+</details>
+
+
+:::tip Additional resources
+
+- [Full source code](https://github.com/MystenLabs/sui/blob/705ee1ed3ce8cfadc4597c6facb6769d7dfb5896/examples/trading/contracts/escrow/sources/shared.move)
+- Concepts: [Shared Objects](../../../concepts/object-ownership/shared)
+- Concepts: [Shared Object Deletion](https://blog.sui.io/ephemeral-shared-objects/)
+- Concepts: [PTBs](../../../concepts/transactions/prog-txn-blocks)
+
+:::
+
+### Testing
+
+Tests for the `escrow` module will be more involved than for `lock` – you will take advantage of `test_scenario`'s ability to simulate multiple transactions from different senders, and interact with shared objects.
+
+You will focus on the test for a successful swap, but a link to all the tests can be found later on.
+
+As with the lock test, start by creating a function to mint a test coin. You also create some constants to represent our transaction senders, `ALICE`, `BOB`, and `DIANE`.
+
+{@inject: examples/trading/contracts/escrow/sources/shared.move#fun=test_coin noComments}
+
+The test body starts with a call to `test_scenario::begin` and ends with a call to `test_scenario::end`. It doesn't matter which address you pass to `begin`, because you will pick one of `ALICE` or `BOB` at the start of each new transaction you write, so set it to `@0x0`:
+
+{@inject: examples/trading/contracts/escrow/sources/shared.move#test}
+
+The first transaction is from `BOB` who creates a coin and locks it. You may remember the ID of the coin and the ID of the key, which you will need later, but now you transfer the locked object and the key itself to `BOB`, because this is what would happen with a real transaction: When simulating transactions in a test, you should only keep around primitive values, not whole objects, which would need to be written to chain between transactions.
```

Comment:
Similar scanning issue for me here:

```suggestion
The first transaction is from `BOB` who creates a coin and locks it. You must remember the ID of the coin and the ID of the key, which you will need later, and then you transfer the locked object and the key itself to `BOB`, because this is what would happen in a real transaction: When simulating transactions in a test, you should only keep around primitive values, not whole objects, which would need to be written to chain between transactions.
```